code/modules/mob/living/simple_animal/hostile/ooze.dm

```diff
@@ -0,0 +1,482 @@
+		return FALSE
+	owner.visible_message("<span class='warning>[ooze] starts attempting to devour [target]!</span>", "<span class='notice'>You start attempting to devour [target].</span>")
+	if(!do_after(ooze, 15, target = ooze.pulling))
+		return FALSE
+	var/mob/living/eat_target = ooze.pulling
+
+	if(!(eat_target.mob_biotypes & MOB_ORGANIC) || eat_target.stat == DEAD)
+		to_chat(src, "<span class='warning'>This creature isn't to my tastes!</span>")
+		return FALSE
+	start_consuming(eat_target)
+
+///Start allowing this datum to process to handle the damage done to  this mob.
+/datum/action/consume/proc/start_consuming(mob/living/target)
+	vored_mob = target
+	vored_mob.forceMove(owner) ///AAAAAAAAAAAAAAAAAAAAAAHHH!!!
+	playsound(owner,'sound/items/eatfood.ogg', rand(30,50), TRUE)
+	owner.visible_message("<span class='warning>[src] devours [target]!</span>", "<span class='notice'>You devour [target].</span>")
+	START_PROCESSING(SSprocessing, src)
+
+///Stop consuming the mob; dump them on the floor
+/datum/action/consume/proc/stop_consuming()
+	STOP_PROCESSING(SSprocessing, src)
+	vored_mob.forceMove(get_turf(owner))
+	playsound(get_turf(owner), 'sound/effects/splat.ogg', 50, TRUE)
+	owner.visible_message("<span class='warning>[owner] pukes out [vored_mob]!</span>", "<span class='notice'>You puke out [vored_mob].</span>")
+	vored_mob = null
+
+///Gain health for the consumption and dump some clone loss on the target.
+/datum/action/consume/process()
+	var/mob/living/simple_animal/hostile/ooze/gelatinous/ooze = owner
+	vored_mob.adjustCloneLoss(6)
+	ooze.heal_ordered_damage((ooze.maxHealth * 0.03), list(BRUTE, BURN, OXY)) ///Heal 6% of these specific damage types each process
+	ooze.adjust_ooze_nutrition(3)
+
+	///Dump em at 200 cloneloss.
+	if(vored_mob.getCloneLoss() >= 200)
+		stop_consuming()
+
+///On owner death dump the current vored mob
+/datum/action/consume/proc/on_owner_death()
+	stop_consuming()
+
+
+///* Gelatinious Grapes code below *\\\\
+
+///Child of the ooze mob which is orientated at being a healer type creature.
+/mob/living/simple_animal/hostile/ooze/grapes
+	name = "Sholean grapes"
+	desc = "A botryoidal ooze from Sholus VII.\nXenobiologists consider it to be one of the calmer and more agreeable species on the planet, but so far little is known about its behaviour in the wild.\nIt undulates in a comforting manner."
+	icon_state = "grapes"
+	icon_dead = "grapes_dead"
+	speed = 1
+	health = 200
+	maxHealth = 200
+	damage_coeff = list(BRUTE = 1, BURN = 0.8, TOX = 0.5, CLONE = 1.5, STAMINA = 0, OXY = 1)
+	melee_damage_lower = 12
+	melee_damage_upper = 12
+	obj_damage = 15
+	deathmessage = "deflates and spills its vital juices!"
+	///The ability lets you envelop a carbon in a healing cocoon. Useful for saving critical carbons.
+	var/datum/action/cooldown/gel_cocoon/gel_cocoon
+	///The ability to shoot a mending globule, a sticky projectile that heals over time.
+	var/obj/effect/proc_holder/globules/globules
```

Comment:
Are these getting cleaned up on mob destruction?

Reply:
doubt it, but it would be better for these guys to listen to the destroy signal of their holder because then its generic behavior.

Reply:
Both work.